### PR TITLE
Transaction process 1/3: Add new schema, migrate data from old schema to new schema

### PR DESCRIPTION
**Pull request by rap1ds**
- [x] Create a TransactionProcess model
- [x] Each community has one TransactionProcess per process type
- [x] Save TransactionProcess in TransactionTypeCreator

### CODE REVIEW

**Comment by Gnito** [comment]
:+1:

**Comment by rap1ds** [comment]
@Gnito Yes they do. When running down-stream migration, how can you tell which were null before running the initial up-stream migration? :) That's the problem.

File: app/controllers/listings_controller.rb [hunk 1]
```diff
@@ -512,7 +512,8 @@ def payment_setup_status(community:, user:, listing:, payment_type:)
     end
   end
 
-  def select_payment_process(price_field:, payment_gateway_available:, preauthorize:)
+  # This is wrong place, yes, but this method will be soon removed
+  def self.select_payment_process(price_field:, payment_gateway_available:, preauthorize:)
```
Comment (ovan) [inline app/controllers/listings_controller.rb]: Why class method?

File: app/services/transaction_type_creator.rb [hunk 2]
```diff
@@ -66,17 +66,26 @@ module TransactionTypeCreator
...
   module_function
 
-  def create(community, transaction_type_class_name)
-    throw "Transaction type #{transaction_type_class_name} not available. Available types are: #{available_types.join(', ')}" unless available_types.include? transaction_type_class_name
+  def create(community, transaction_type_class_name, process)
+    throw "Transaction type '#{transaction_type_class_name}' not available. Available types are: #{available_types.join(', ')}" unless available_types.include? transaction_type_class_name
+    throw "Transaction process '#{process}' not available. Available processes are: #{TRANSACTION_PROCESSES.join(', ')}" unless TRANSACTION_PROCESSES.include? process.to_sym
+
+    author_is_seller = transaction_type_class_name != "Request"
+    transaction_process = TransactionProcess.where(community_id: community.id, author_is_seller: author_is_seller, process: process).first_or_create
```
Comment (ovan) [inline app/services/transaction_type_creator.rb]: Is there a bang version of first_or_create?

File: db/migrate/20150303140556_add_transaction_process_id_to_transaction_type.rb [hunk 3]
```diff
@@ -0,0 +1,10 @@
+class AddTransactionProcessIdToTransactionType < ActiveRecord::Migration
+  def up
+    add_column :transaction_types, :transaction_process_id, :int, after: :community_id
+    add_index :transaction_types, :transaction_process_id
```
Comment (ovan) [inline db/migrate/20150303140556_add_transaction_process_id_to_transaction_type.rb]: What do we need this index for? What's the use case for looking up a transaction type by transaction process?

File: app/services/transaction_type_creator.rb (hunk 2 above)
Comment (rap1ds) [inline app/services/transaction_type_creator.rb]: Yes there is!

**Pull request merged**